pallets/dapp-staking-v3/src/migration.rs

```diff
@@ -56,33 +56,60 @@ pub mod versioned_migrations {
 // TierThreshold as percentage of the total issuance
 mod v8 {
     use super::*;
+    use crate::migration::v7::TierParameters as TierParametersV7;
     use crate::migration::v7::TiersConfiguration as TiersConfigurationV7;
 
-    pub struct VersionMigrateV7ToV8<T>(PhantomData<T>);
+    pub struct VersionMigrateV7ToV8<T, TierThresholds>(PhantomData<(T, TierThresholds)>);
```

Comment:
```suggestion
    pub struct VersionMigrateV7ToV8<T, TierThresholds> { tier_thresholds: BoundedVec<TierThreshold, T::NumberOfTiers>, ...PhantomData};
```
Passing `tier_thresholds` as argument may be a better idea

Reply:
I am not sure to understand the benefit here apart more clarity.

I can always create a constructor/getter for `VersionMigrateV7ToV8` struct like:

```rust
pub struct VersionMigrateV7ToV8<T: Config, TierThresholds> {
        tier_thresholds: BoundedVec<TierThreshold, T::NumberOfTiers>,
        _phantom: PhantomData<(T, TierThresholds)>,
    }

impl<T: Config, TierThresholds: Get<[TierThreshold; 4]>> VersionMigrateV7ToV8<T, TierThresholds> {
    pub fn new() -> Self {
        let tier_thresholds: BoundedVec<TierThreshold, T::NumberOfTiers> =
            BoundedVec::try_from(TierThresholds::get().to_vec()).unwrap();
        Self {
            tier_thresholds,
            _phantom: PhantomData,
        }
    }
}
```

but a variable passed via *parameter_types!* will still be required for the `TierThresholds` type in the runtime migration side. Is this what you have in mind?

Reply:
my idea was to pass it as parameter `v8::VersionMigrateV7ToV8(tier_thresholds)`